node/events_test.ts

```diff
@@ -0,0 +1,1138 @@
+import { assert } from "../testing/asserts.ts";
+import { EventEmitter, getEventListeners, on, once } from "./events.ts";
+import { assertCallbackErrorUncaught } from "./_utils.ts";
+import type { EventListenerT } from "./events.ts";
+import { expectType } from "https://raw.githubusercontent.com/TypeStrong/ts-expect/736658bd9f1c23ad9e2676c27e9bdc9297309ae9/src/index.ts";
+import type { TypeEqual } from "https://raw.githubusercontent.com/TypeStrong/ts-expect/736658bd9f1c23ad9e2676c27e9bdc9297309ae9/src/index.ts";
+
+// TODO: 1. Those tests don't do much while running tests, they aren't testing the implementation but types added on top
+//          of mjs code, so their focus is to fail at compile-time when types are invalid
+//          Maybe exclude it to separate task taking <filename>_test_types.ts?
+// TODO: 2. Create a module in Deno.test for expecting types:
+//          Issue link https://github.com/denoland/deno_std/issues/2145
```

Comment:
Testing of types is totally fine! Let's keep discussing public type assertion APIs in #2145